ext/ffi/ffi.c

```diff
@@ -4697,21 +4779,19 @@ ZEND_MINIT_FUNCTION(ffi)
 	zend_ffi_cdata_value_handlers.get_constructor      = zend_fake_get_constructor;
 	zend_ffi_cdata_value_handlers.free_obj             = zend_ffi_cdata_free_obj;
 	zend_ffi_cdata_value_handlers.clone_obj            = zend_ffi_cdata_clone_obj;
-	zend_ffi_cdata_value_handlers.read_property        = zend_fake_read_property;
-	zend_ffi_cdata_value_handlers.write_property       = zend_fake_write_property;
+	zend_ffi_cdata_value_handlers.read_property        = zend_ffi_cdata_get;
+	zend_ffi_cdata_value_handlers.write_property       = zend_ffi_cdata_set;
 	zend_ffi_cdata_value_handlers.read_dimension       = zend_fake_read_dimension;
 	zend_ffi_cdata_value_handlers.write_dimension      = zend_fake_write_dimension;
 	zend_ffi_cdata_value_handlers.get_property_ptr_ptr = zend_fake_get_property_ptr_ptr;
-	zend_ffi_cdata_value_handlers.get                  = zend_ffi_cdata_get;
-	zend_ffi_cdata_value_handlers.set                  = zend_ffi_cdata_set;
 	zend_ffi_cdata_value_handlers.has_property         = zend_fake_has_property;
```

Comment:
The `cdata` property should also be handled by has_property.

Reply:
FFI::CData doesn't provide has_propery even for structures and unions. So, no necessary for scalars.

Reply:
Is there a reason why this is not provided, or just not implemented yet?

Reply:
C doesn't provide isset() or iterators for struct/union. FFI shouldn't as well.

Reply:
C doesn't have isset(), so it can't provide it ^^ But in PHP, if `$obj->prop` access is supported, then `isset($obj->prop)` should also be supported.

Reply:
It's also not just about isset() itself, but any operations based on it. E.g the `??` operator does not work correctly if there is no correct has_property implementation.

Reply:
From https://www.php.net/manual/en/class.ffi-cdata.php

Note: Noteable limitations are that FFI\CData instances do not support isset(), empty() and unset(), and that wrapped C structs and unions do not implement Traversable.

I think, this functionality is useless (exception is thrown).